solutions/devsprint-bruno-faganello-2/SpaceApp/Screens/NextLaunchView.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  NextLaunchView.swift
+//  SpaceApp
+//
+//  Created by Paolo Prodossimo Lopes on 26/04/22.
+//
+
+import SwiftUI
+
+struct NextLaunchView: View {
+    
+    let viewData: NextLaunchViewData
+    
+    var body: some View {
+        ZStack {
+            Color.black
+            
+            VStack(spacing: 16) {
+                HStack(spacing: 16) {
+                    bagdeIconImage
+                    launchInformationsStack
+                }
+                
+                Text(viewData.description)
+                    .lineLimit(3)
+                    .font(.system(size: 20, weight: .regular))
+                    .foregroundColor(.gray)
+            }
+            .padding(16)
+            .frame(maxWidth: .infinity)
+            .background(.white.opacity(0.1))
+            .cornerRadius(10)
+        }
+    }
+    
+    private var bagdeIconImage: some View {
+        AsyncImage(url: viewData.badgeURL) { phase in
+            switch phase {
+            case .empty:
+                ProgressView()
+                    .progressViewStyle(.circular)
+            case .success(let image):
+                image
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+            default:
+                Text("Failed fetching image. Make sure to check your data connection and try again.")
+            }
+        }
+        .frame(width: 125, height: 125)
```

Comment:
[REQUEST CHANGED]

This frame need to be inside 
```
image
 .resizable()
.aspectRatio(contentMode: .fit)
.frame(width: 125, height: 125)
```

https://wwdcbysundell.com/2021/using-swiftui-async-image/

Reply:
I set outside closure, because the frame will be applied on whole AsyncImage (image and was is empty case with LoaderView)